### DISCUSSION TITLE
Manual Upgrade Flags

### DISCUSSION THREAD

**Discussion by joshpaulchan** [k3s-io/k3s#8721] [Q&A]
Hi! Running k3s has been pretty smooth so far. However on reboot my `server` node is failing to start, and I think I've traced it down to an older version of flannel, so I've been attempting to upgrade k3s to fix that. 

Looking at the [manual upgrade docs](https://docs.k3s.io/upgrades/manual#upgrade-k3s-using-the-installation-script), it suggests running:

```
curl -sfL https://get.k3s.io | <EXISTING_K3S_ENV> sh -s - <EXISTING_K3S_ARGS>
```

I can't find documentation on what the value of `EXISTING_K3S_ENV` is supposed to be, nor does a short reading of the install script seem to reveal anything. What am I missing here?

Re-running the install script by itself seems to get as far as downloading the k3s hash, but exits:
```
root@k3s-master:~# curl -sfL https://get.k3s.io | sh -
[INFO]  Finding release for channel stable
curl: (7) Couldn't connect to server
[INFO]  Using v1.27.6+k3s1 as release
[INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/v1.27.6+k3s1/sha256sum-amd64.txt
```

**Comment by brandond** [comment]
`EXISTING_K3S_ENV` and `EXISTING_K3S_ARGS` would be the same environment variables and arguments as you used when initially installing K3s. If you don't provide the same options when re-installing, your previous configuration will be lost, as the installer rewrites the env file and systemd unit every time it is run.

The two errors from the script (`curl: (7) Couldn't connect to server`, and failure to download the checksum file) indicate that your host is having trouble reaching both our channel server at update.k3s.io, and github.com. Do you need to configure a proxy server or something?

**Reply by joshpaulchan** [reply]
Sorry for the long follow up! I got past this issue.

I'm not sure what the `curl: (7)` error was but I think it was primarily? a red herring. It was at `1.25.x` previously and fetching new details from the stable channel (the updated `1.27.x` version in this case) seemed to work fine - what ultimately got this working was I had to run the `k3s-killall` script and then attempt the upgrade, which then successfully completed.

**Reply by pauldsmyth** [reply]
I'm attempting to upgrade a k3s installation but I installed using the following:
"curl -sfL https://get.k3s.io | sh -"
I didn't set any env vars or args. Surely these should be written to a file for retrieval?

**Reply by brandond** [reply]
If you didn't set any vars or args, I'm not sure what you're expecting to see written to a file? The install script writes the default.env file, and the systemd unit file. If you didn't pass any extra flags to the install script, then the only command in the unit will be `k3s server` or `k3s agent` with no extra args.

**Comment by tim-oe** [comment]
stumbled on this as well, other than the node settings i used the standard install, no args running upgrade and it seems to have overwrote the config..yml file on the controller.
ctlr:
`curl -sfL https://get.k3s.io | sh -`
nodes:
curl -sfL https://get.k3s.io | K3S_URL=https:controller.host.net:6443 K3S_TOKEN=??? sh -

it did not overwrite the token at least

**Reply by brandond** [reply]
The install script does not alter config.yaml. It will overwrite the default .env file, and the systemd unit, but not the config file.
